ts/api/pagopa.ts

```diff
@@ -107,12 +172,101 @@ export const PagoPaClient = (
     )
   });
 
+  const checkPayment: (
+    pagoPaToken: string
+  ) => CheckPaymentType = pagoPaToken => ({
+    method: "get",
+    url: ({ paymentId }) => `/v1/payments/${paymentId}/actions/check`,
+    query: () => ({}),
+    headers: AuthorizationBearerHeaderProducer(pagoPaToken),
+    response_decoder: basicResponseDecoderWith401AndCast<PaymentResponse>(
+      PaymentResponse
+    )
+  });
+
+  const getPspList: (pagoPaToken: string) => GetPspsListType = pagoPaToken => ({
+    method: "get",
+    url: () => "/v1/psps",
+    query: ({ paymentId }) => ({
+      paymentType: "CREDIT_CARD",
+      idPayment: paymentId
+    }),
+    headers: AuthorizationBearerHeaderProducer(pagoPaToken),
+    response_decoder: basicResponseDecoderWith401AndCast<PspListResponse>(
+      PspListResponse
+    )
+  });
+
+  const updateWalletPsp: (
+    pagoPaToken: string
+  ) => UpdateWalletPspType = pagoPaToken => ({
+    method: "put",
+    url: ({ walletId }) => `/v1/wallet/${walletId}`,
+    query: () => ({}),
+    body: ({ pspId }) =>
+      JSON.stringify({
+        data: {
+          type: "CREDIT_CARD",
+          idPsp: pspId
+        }
+      }),
+    headers: composeHeaderProducers(
+      AuthorizationBearerHeaderProducer(pagoPaToken),
+      ApiHeaderJson
+    ),
+    response_decoder: basicResponseDecoderWith401AndCast<WalletResponse>(
+      WalletResponse
+    )
+  });
+
+  const postPayment: (
+    pagoPaToken: string
+  ) => PostPaymentType = pagoPaToken => ({
+    method: "post",
+    url: ({ paymentId }) => `/v1/payments/${paymentId}/actions/pay`,
+    query: () => ({}),
+    body: ({ walletId }) =>
+      JSON.stringify({
+        data: {
+          tipo: "web",
+          idWallet: `${walletId}`
+        }
+      }),
+    headers: composeHeaderProducers(
+      AuthorizationBearerHeaderProducer(pagoPaToken),
+      ApiHeaderJson
+    ),
+    response_decoder: basicResponseDecoderWith401AndCast<TransactionResponse>(
+      TransactionResponse
+    )
+  });
+
   return {
-    getSession: (walletToken: string) =>
-      createFetchRequestForApi(getSession, options)({ token: walletToken }),
+    walletToken,
+    getSession: (
```

Comment:
instead of passing the `pagoPaToken` to all the other methods, you can split the client in two and have `getSession` return an "authorized" client that embed the `pagopaToken`

Reply:
In order to keep the things going, I would suggest we carry on with this PR & have this recommendation for a separate authorized client be a technical debt to be handled down the line